### DISCUSSION TITLE
Shadow database concept doesn't work for pre-existing production databases?

### DISCUSSION THREAD

**Discussion by JVMartin** [prisma/prisma#13459] [Q&A] [answered]
I had an existing production database built with TypeORM, that I have cleanly transitioned to Prisma.

My initial migration, therefore, looks like this, dropping TypeORM tables, and renaming tables to match Prisma conventions:
```sql
--- prisma/migrations/20220222031047_init/migration.sql
---
--- Initial migrations to transform us from TypeORM convention schema to Prisma convention schema
--- Testing in QA, then applying to production

DROP TABLE "migrations";
DROP TABLE "typeorm_metadata";

ALTER TABLE "config"
RENAME TO "configs";

ALTER TABLE "lead"
RENAME TO "leads";

ALTER TABLE "permission"
RENAME TO "permissions";

ALTER TABLE "refresh-token"
RENAME TO "refresh_tokens";

ALTER TABLE "role"
RENAME TO "roles";

...
```
(Truncated for brevity)

This worked fine and dandy!

However, now when I go to add a 2nd migration, I'm unable to because the shadow database starts empty, and it can't drop the tables that don't exist, rename tables that don't exist, etc. from the init migration:
```
» npx prisma migrate dev --name add_config --create-only
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "xxx", schema "public" at "xxx.b.db.ondigitalocean.com:25060"

Error: P3006

Migration `20220222031047_init` failed to apply cleanly to the shadow database. 
Error code: P1014
Error:
The underlying table for model `migrations` does not exist.
```

I'm at a total loss for how to proceed.

Also, what if I wanted to delete rows from prod DB in a migration?? Those won't exist in shadow DB either. How can I create and use Prisma migrations in production? My migrations should *only* apply to production, never to dev environments since those are always totally cleanly synced and seeded.

**Comment by janpio** [comment, marked as answer]
`migrate dev` usually requires to start from nothing, and if that is not the case that state is emulated by a process we call "baselining": https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining#why-you-need-to-baseline

**Reply by JVMartin** [reply]
Thank you so much, somehow I never found this when scouring the docs for this issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
